### PR TITLE
alternative release_model with release branch and dev branch

### PR DESCRIPTION
**Pull request by Glease**
so I have been revisiting our release model recently (actually, not *very* recently, just about 10 months ago). I strongly feels the need for a two branched model. Here I propose this model

the original proposal was intended for posting on discord, but this turns out to be a little too lengthy for it, and I don't really want to postpone this any more just to reformat the doc, so I just stuffed it here and in txt format.

# DO NOT MERGE BEFORE WE AGREE ON THIS

### CODE REVIEW

**Review by bombcar — APPROVED** [review]

File: governance/release_model.txt [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+* https://github.com/GTNewHorizons/GT5-Unofficial/pull/1627
+
+---
+
+**Modpack Release Model**
+
+* for versions from backport branch (__stable patch release__), it will be 2.4.1, 2.4.2, etc. every release @ everyone and (probably) will be on curse.
+* for versions from dev branch (__dev release__), it will be 2.5.0-xxxxx.1, where xxxxx can be alpha, beta, or RC.
+  * alpha is first stage. 2.5.0-alpha.1, 2.5.0-alpha.2 etc. alpha version mentions nerf nerds and zeta test group only. this is the old "dev build", and should be the build we release for the 90% of the time.
+  * we bump it to 2.4.0-beta.1 when we feels next stable should be made. first beta should @ everyone to encourage people to upgrade and test. later beta will @ nerf nerds and zeta test group only. beta stage usually last 2 weeks.
```
[the remark this answers is not in the document]
Comment (Glease) [reply governance/release_model.txt:43]: eh yeah. this was mostly written between 2.3.0 and 2.4.0 so I might have mixed it up

[the remark this answers is not in the document]
Comment (miozune) [reply governance/release_model.txt]: 2.54.0 now lol

File: governance/release_model.txt [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+**New development workflow**
+
+1. developer submit PR **against master branch** (as usual). 
+2. PR review, address reviews.
+3. Merge or close.
+**(Below is the addition)**
+4. any GTNH dev or original developer put a special label on the same PR to propose a cherrypick to a specific release version.
+5. A github actions bot kick in to attempt an automatic cherrypick to that branch in a new PR (like the automatic spotless apply PR). This new PR will automatically put @GTNewHorizons/core on reviewer list and add a label of "cherrypick" (for easy filtering).
+  5.1 probably need to make the bot auto apply spotless after cherrypick too
```
[the remark this answers is not in the document]
Comment (Glease) [reply governance/release_model.txt:79]: wtf I'd blame this lame keyboard

File: governance/release_model.txt [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+6. @core team review. merge or close.
```
Comment (miozune) [inline governance/release_model.txt:80]: core team doesn't exist now

**Review by miozune — COMMENTED** [review]
Is it intended this file is .txt instead of .md?

**Comment by Dream-Master** [comment]
@Glease

**Review by Caedis — APPROVED** [review]

**Review by boubou19 — APPROVED** [review]
i'm ok with this change, but this will require tons of work to implement, so i guess we will start working on it in 2024. We also need to shorten the time between 2 stable releases btw.

**Pull request merged**